libcontainer/utils/utils.go

```diff
@@ -36,10 +39,27 @@ func ResolveRootfs(uncleanRootfs string) (string, error) {
 }
 
 // ExitStatus returns the correct exit status for a process based on if it
-// was signaled or exited cleanly.
+// was signaled or exited cleanly
 func ExitStatus(status syscall.WaitStatus) int {
 	if status.Signaled() {
 		return exitSignalOffset + int(status.Signal())
 	}
 	return status.ExitStatus()
 }
+
+// WriteJSON writes the provided struct v to w using standard json marshaling
+func WriteJSON(w io.Writer, v interface{}) error {
+	data, err := json.Marshal(v)
+	if err != nil {
+		return err
+	}
+	buf := bytes.NewBuffer(data)
+	n, err := buf.WriteTo(w)
+	if err != nil {
+		return err
+	}
+	if int(n) != len(data) {
+		return fmt.Errorf("object not written")
```

Comment:
would be nice to have the returned length and expected value here, something like: `fmt.Errorf("object not written, got(%d), expected(%d)", n, len(data))`